src/get_climate_future.R

```diff
@@ -0,0 +1,184 @@
+        
+      clim <- climateR::getMACA(AOI = aoi, 
+                              varname = vars,#,"pet","vs"),
+                              model = models[j],
+                              scenario = scenarios[j],
+                              startDate = start,
+                              endDate = end) 
+        
+      # If the result is a spatraster (i.e., AOI spanned more than 1 grid cell)
+      if(is.list(clim) | inherits(clim[[1]], "SpatRaster")) {
+        
+        clim_crs <- st_crs(clim[[1]])
+    
+        if(st_crs(clim[[1]]) != st_crs(sf)) {
+          
+          clim <- clim %>%
+            purrr::map(
+              # getGridMET defaults AOI to bbox - so crop / mask results to sf extent
+              ~terra::crop(., st_transform(aoi, crs = clim_crs), mask = TRUE),
+              crs = clim_crs)
+          } else {
+            
+            clim <- clim %>%
+              purrr::map(
+                # defaults AOI to bbox - so crop / mask results to sf extent
+                ~terra::crop(., aoi %>% 
+                               st_transform(., crs = clim_crs), 
+                             mask = TRUE),
+                crs = clim_crs)
+            
+          }
+    
+          all_climate_data[[idx]] <- clim %>%
+            purrr::map_dfr(~ as.data.frame(., xy = TRUE)) %>%
+            data.table() %>%
+            pivot_longer(-(x:y),
+                         names_to = "var_temp",
+                         values_to = "val") %>%
+            separate_wider_delim(var_temp, "_", 
+                                 names = c("var","date","GCM",
+                                           "Ensemble","RCP")) %>%
+            drop_na(val) %>%
+            group_by(x, y, date) %>%
+            pivot_wider(names_from = "var", 
+                        values_from = "val") %>%
+            dplyr::mutate(date = as.Date(date),
+                   ppt_mm = pr,
+                   tmax_C = tasmax - 273.15,
+                   tmin_C = tasmin - 273.15,
+                   tmean_C = (tmax_C + tmin_C)/2,
+                   join_index = aoi$join_index) %>%
+            dplyr::select(-c("tasmax","tasmin","pr"))
+    
+    } else {
+      
+      all_climate_data[[idx]] <- clim %>%
+        data.table() %>%
+        # since polygon grabbed a single grid, does not provide the coordinates
+        # of the gridMET cell, so we fill in x and y with the coordinates
+        # of the sf object:
+        dplyr::mutate(x = sf::st_coordinates(aoi)[[1]],
+                      y = sf::st_coordinates(aoi)[[2]]) %>%
+        # Then do all other cleaning steps done for polygon sf objects:
+        dplyr::mutate(date = as.Date(date),
+                      ppt_mm = pr,
+                      tmax_C = tasmax - 273.15,
+                      tmin_C = tasmin - 273.15,
+                      tmean_C = (tmax_C + tmin_C)/2,
+                      join_index = aoi$join_index) %>%
+        dplyr::select(-c("tasmax","tasmin","pr"))
```

Comment:
```suggestion
       all_climate_data[[i]] <- clim %>%
          data.table() %>%
          # names of columns include va_mode_rcp so must rename
          rename_with(~ str_split(.x, "_", n = 2) %>% map_chr(1)) %>%
          # since point pulls from gridMET do not provide the coordinates
          # of the gridMET cell, we fill in x and y with the coordinates
          # of the sf object:
          dplyr::mutate(x = sf::st_coordinates(aoi)[[1]],
                        y = sf::st_coordinates(aoi)[[2]]) %>%
          # Then do all other cleaning steps done for polygon sf objects:
          dplyr::mutate(date = as.Date(date),
                        ppt_mm = pr,
                        tmax_C = tasmax - 273.15,
                        tmin_C = tasmin - 273.15,
                        tmean_C = (tmax_C + tmin_C)/2,
                        join_index = aoi$join_index,
                        GCM = models[j],
                        RCP = scenarios[j],
                        Ensemble = "r1i1p1") %>%
          dplyr::select(-c("tasmax","tasmin","pr")) 
```